lib/proxy/validations.rb

```diff
@@ -11,20 +18,42 @@ def valid_mac?(mac)
     false
   end
 
+  def valid_ip?(ip)
+    !!IPAddr.new(ip) rescue false
```

Comment:
`IPAddr.new` also accepts a subnet so it'll accept `192.0.2.0/24`. Perhaps:
```suggestion
    valid_ip4?(ip) || valid_ip6?(ip)
```